src/testing/fake_registry/fake_registry.go

```diff
@@ -0,0 +1,109 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package main is fake HTTP registry server. See README.md for more details.
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	pbp "github.com/lowRISC/opentitan-provisioning/src/proxy_buffer/proto/proxy_buffer_go_pb"
+
+	"google.golang.org/grpc/codes"
+)
+
+var (
+	port                   = flag.Int("port", 9999, "Port to listen, defaults to 9999")
+	registerDeviceURL      = flag.String("register_device_url", "/registerDevice", "URL to listen to RegisterDevice requests. Defaults to '/registerDevice'")
+	batchRegisterDeviceURL = flag.String("batchRegister_device_url", "/batchRegisterDevice", "URL to listen to BatchRegisterDevice requests. Defaults to '/batchRegisterDevice'")
+)
+
+type callError struct {
+	Code codes.Code `json:"code"`
+}
+
+type registerResponse struct {
+	DeviceID string     `json:"deviceId"`
+	Error    *callError `json:"error"`
+}
+
+type batchRegisterResponse struct {
+	Responses []*registerResponse `json:"responses"`
+}
+
+var callErrorOK = callError{Code: codes.OK}
+
+func handleError(w http.ResponseWriter, errorMessage string) {
+	w.Write([]byte(errorMessage))
+	w.WriteHeader(http.StatusBadRequest)
+}
+
+func registerDevice(w http.ResponseWriter, r *http.Request) {
+	reqBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		handleError(w, "failed to read request body")
+		return
+	}
+	req := &pbp.DeviceRegistrationRequest{}
+	if err := json.Unmarshal(reqBytes, req); err != nil {
```

Comment:
it might be cool (in a follow up PR) to add logic for verifying the payload signatures in the fake registry too. as a stretch goal.

Reply:
Good point! Created #171 to track this.